bindings/cs/MongoDB.Libmongocrypt/CryptOptions.cs

```diff
@@ -24,18 +26,18 @@ namespace MongoDB.Libmongocrypt
     /// </summary>
     public class CryptOptions
     {
-        public IReadOnlyDictionary<KmsType, IKmsCredentials> KmsCredentialsMap { get; }
+        public IReadOnlyList<KmsCredentials> KmsCredentials { get; }
         public byte[] Schema { get; }
 
-        public CryptOptions(IDictionary<KmsType, IKmsCredentials> map) : this(map, null)
+        public CryptOptions(IEnumerable<KmsCredentials> credentials) : this(credentials, null)
         {
         }
 
         public CryptOptions(
-            IDictionary<KmsType, IKmsCredentials> map,
+            IEnumerable<KmsCredentials> credentials,
             byte[] schema)
         {
-            KmsCredentialsMap = new ReadOnlyDictionary<KmsType, IKmsCredentials>(map);
+            KmsCredentials = new ReadOnlyCollection<KmsCredentials>((credentials ?? throw new ArgumentNullException(nameof(credentials))).ToList());
```

Comment:
Nit: `?? throw new ArgumentNullException(nameof(credentials))` is less readable than validating argument in at the top of the method, so I'd suggest the latter.

Reply:
I agree in general, but for that, we will need to introduce a new variable or `if` condition which will make it less readable as well. So since it's `nit`, I would leave it as is